tests/check_check_sub.c

```diff
@@ -443,6 +443,151 @@ START_TEST(test_ck_assert_uint_expr)
   ck_assert_uint_eq(x, y);
 } END_TEST
 
+START_TEST(test_ck_assert_floating_eq)
+{
+  double x = 1.1;
+  double y = 1.1;
+  ck_assert_floating_eq(x, y);
+  y+=0.1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_eq(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ne)
+{
+  double x = 1.1;
+  double y = 1.2;
+  ck_assert_floating_ne(x, y);
+  y = x;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ne(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ne_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ne(1%d, 1%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_lt)
+{
+  double x = 2.0;
+  double y = 2.5;
+  ck_assert_floating_lt(x, y);
+  y-=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_lt(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_lt_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_lt(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le)
+{
+  double x = 2.0;
+  double y = 2.5;
+  ck_assert_floating_le(x, y);
+  ck_assert_floating_le(x, x);
+  y-=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_le_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_le(3%d, 2%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt)
+{
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_gt(x, y);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_gt_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_gt(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge)
+{
+  double x = 2.5;
+  double y = 2.0;
+  ck_assert_floating_ge(x, y);
+  ck_assert_floating_ge(x, x);
+  y+=1;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(x, y);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_ge_with_mod)
+{
+  int d = 2;
+  int f = 2;
+  record_failure_line_num(__LINE__);
+  ck_assert_floating_ge(2%d, 3%f);
+}
+END_TEST
+
+START_TEST(test_ck_assert_floating_eq_prec)
```

Comment:
When testing this method should data available in [cfloat.h](http://www.cplusplus.com/reference/cfloat/) be used to determine the correctness? This will result in more extreme examples that check boundary conditions.